lib/spack/spack/fetch_strategy.py

```diff
@@ -337,7 +337,7 @@ def fetch(self):
         url = None
         errors = []
         for url in self.candidate_urls:
-            if not self._existing_url(url):
+            if not web_util.url_exists(url, self.curl):
```

Comment:
I think using `self.curl` triggers:

https://github.com/spack/spack/blob/99fcc5760755e41f2af5431bcafc4fd3f326f47d/lib/spack/spack/fetch_strategy.py#L295-L302

which in turn has the `required=True` argument. I didn't test it, but by just reading the code I think this might be the cause of the behavior reported in https://github.com/spack/spack/pull/34029#issuecomment-1332566793 @trws @tldahlgren

Reply:
If the above is correct, this PR makes `curl` a requirement even if Spack is configured not to use it.

Reply:
🤦 Great catch.